addons/auth_ldap/models/res_company_ldap.py

```diff
@@ -79,19 +79,18 @@ def _connect(self, conf):
         return connection
 
     def _get_entry(self, conf, login):
-        filter, dn, entry = False, False, False
-        try:
-            filter = filter_format(conf['ldap_filter'], (login,))
-        except TypeError:
-            _logger.warning('Could not format LDAP filter. Your filter should contain one \'%s\'.')
-        if filter:
-            results = self._query(conf, tools.ustr(filter))
-
-            # Get rid of (None, attrs) for searchResultReference replies
-            results = [i for i in results if i[0]]
-            if len(results) == 1:
-                entry = results[0]
-                dn = results[0][0]
+        dn, entry = False, False
+        filter = filter_format(
+            conf['ldap_filter'],
+            (login,) * conf['ldap_filter'].count('%s')
```

Comment:
You can kill this with just setting `%%s` in your filter, although I'm not sure why would you want to do that.

Instead, it'd be better to allow formatting using a dict, so you can use as many occurrences of `%(login)s` as you need, or you could even add things such as `%(email)s` also.

A backwards-compatibility layer could be added too. I'm not sure if this should go for master though.

Reply:
Thank you for your review.

I agree that a `%(login)s` based solution would be the best. Here I wanted to keep the patch as small as possible, but I am ok to provide another more complete patch. If not in master, where should this be implemented in your opinion?

As for this patch, what would you need me to do to approve it?

Reply:
@Yajo Would you consider approving this patch?

Reply:
It's just a comment, my approval here is pointless, you have to convince Odoo people 😅

Reply:
I thought two approvals from anyone were required, but maybe this just applies for OCA.

Reply:
Yes, that is OCA, not here